pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -257,3 +272,55 @@ func (c *cephStatusChecker) conditionMessageReason(condition cephv1.ConditionTyp
 
 	return condition, reason, message
 }
+
+// forceDeleteStuckPodsOnNotReadyNodes lists all the nodes that are in NotReady state and
+// gets all the pods on the failed node and force delete the pods stuck in terminating state.
+func forceDeleteStuckPodsOnNotReadyNodes(context *clusterd.Context, namespace, clusterName string) error {
+	nodes, err := k8sutil.GetNotReadyKubernetesNodes(context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to get NotReady nodes")
+	}
+	for _, node := range nodes {
+		pods, err := getPodsOnNode(context, node.Name, namespace, clusterName)
+		if err != nil {
+			logger.Errorf("failed to get pods on NotReady node %q. %v", node.Name, err)
+		}
+		for _, pod := range *pods {
+			if err := k8sutil.ForceDeletePodIfStuck(context, pod); err != nil {
+				logger.Errorf("skipping forced delete of stuck pod %q. %v", pod.Name, err)
+			}
+		}
+	}
+	return nil
+}
+
+func getPodsOnNode(ctx *clusterd.Context, node, namespace, clusterName string) (*[]v1.Pod, error) {
+	labels := []string{
+		fmt.Sprintf("rook_cluster=%s", clusterName),
+		"driver=ceph-csi",
+		"app=rook-ceph-operator",
+	}
+	listOpts := metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.nodeName=%s", node),
+	}
+	podsOnNode := []v1.Pod{}
+	for _, label := range labels {
+		listOpts.LabelSelector = label
+		pods, err := ctx.Clientset.CoreV1().Pods(namespace).List(context.TODO(), listOpts)
+		if err != nil {
+			return &podsOnNode, errors.Wrapf(err, "failed to get pods on node %q", node)
+		}
+		podsOnNode = append(podsOnNode, pods.Items...)
+	}
+	return &podsOnNode, nil
```

Comment:
Why return the reference?
```suggestion
	return podsOnNode, nil
```